lang/src/org/partiql/lang/ast/passes/StatementRedactor.kt

```diff
@@ -82,238 +73,256 @@ fun redact(statement: String,
+ * [providedSafeFieldNames] is an optional set of fields whose values are not to be redacted. If no set is provided,
+ * all literals will be redacted.
+ *     For example, given a [providedSafeFieldNames] of Set('hashkey')
+ *     The query of `SELECT * FROM tb WHERE hashkey = 'a' AND attr = 12345` will be redacted to:
+ *                  `SELECT * FROM tb WHERE hashkey = 'a' AND attr = ***(Redacted)`
+ * [userDefinedFunctionRedactionConfig] is an optional mapping of UDF names to functions determining which call
+ * arguments are to be redacted. For an example, please check StatementRedactorTest.kt for more details.
  */
 fun redact(statement: String,
            ast: ExprNode,
            providedSafeFieldNames: Set<String> = emptySet(),
            userDefinedFunctionRedactionConfig: Map<String, UserDefinedFunctionRedactionLambda> = emptyMap()): String {
-    safeFieldNames = providedSafeFieldNames
 
-    val statementRedactionRewriter = StatementRedactionRewriter(statement, ast, userDefinedFunctionRedactionConfig)
-    statementRedactionRewriter.rewriteExprNode()
-    return statementRedactionRewriter.getRedactedStatement()
+    val partiqlAst = ast.toAstStatement()
+    val statementRedactionVisitor = StatementRedactionVisitor(statement, providedSafeFieldNames, userDefinedFunctionRedactionConfig)
+    statementRedactionVisitor.walkStatement(partiqlAst)
+    return statementRedactionVisitor.getRedactedStatement()
 }
 
 /**
- * Redact sensitive data [Literal] not assigned with [safeFieldNames] to "***(Redacted)"
- *
- * TODO: when migrating to PIG, please use PartiqlAst.Visitor and not PartiqlAst.VisitorTransform!"
+ * Redact [PartiqlAst.Expr.Lit]s not assigned with [safeFieldNames] to "***(Redacted)". Function calls that have an
+ * entry in [userDefinedFunctionRedactionConfig] will have their arguments redacted based on the redaction lambda.
  */
-private class StatementRedactionRewriter(
-        private val statement: String,
-        node: ExprNode,
-        private val userDefinedFunctionRedactionConfig: Map<String, UserDefinedFunctionRedactionLambda>
-) : AstRewriterBase() {
-    private val root = node
+private class StatementRedactionVisitor(
+    private val statement: String,
+    private val safeFieldNames: Set<String>,
+    private val userDefinedFunctionRedactionConfig: Map<String, UserDefinedFunctionRedactionLambda>
+) : PartiqlAst.Visitor() {
     private val sourceLocationMetaForRedaction = arrayListOf<SourceLocationMeta>()
     private val maskPattern = "***(Redacted)"
 
-    fun rewriteExprNode(): ExprNode {
-        return super.rewriteExprNode(root)
-    }
-
-    override fun rewriteSelectWhere(node: ExprNode): ExprNode {
-        redactOnField(node)
-        return super.rewriteSelectWhere(node)
+    override fun visitExprSelect(node: PartiqlAst.Expr.Select) {
+        if (node.where != null) {
+            redactExpr(node.where)
```

Comment:
nit: `node.where?.let { redactExpr(it) }` is the same but even shorter.

Could apply this to all similar `if` statements below.

Reply:
Thanks. Changed